runtime/src/main/java/com/flipkart/flux/controller/WorkFlowExecutionController.java

```diff
@@ -13,32 +13,122 @@
 
 package com.flipkart.flux.controller;
 
+import com.flipkart.flux.api.EventData;
+import com.flipkart.flux.dao.iface.EventsDAO;
+import com.flipkart.flux.dao.iface.StateMachinesDAO;
 import com.flipkart.flux.domain.Context;
+import com.flipkart.flux.domain.Event;
 import com.flipkart.flux.domain.State;
 import com.flipkart.flux.domain.StateMachine;
+import com.flipkart.flux.exception.IllegalEventException;
+import com.flipkart.flux.exception.IllegalStateMachineException;
+import com.flipkart.flux.impl.RAMContext;
 
-import java.util.List;
+import javax.inject.Inject;
+import java.util.HashSet;
+import java.util.Set;
 
 /**
- * @understands Controls the execution flow of a given state machine
+ * <code>WorkFlowExecutionController</code> controls the execution flow of a given state machine
+ * @author shyam.akirala
  */
-public interface WorkFlowExecutionController<T> {
+public class WorkFlowExecutionController {
+
+    @Inject
+    StateMachinesDAO stateMachinesDAO;
+
+    @Inject
+    EventsDAO eventsDAO;
+
     /**
-     * Perform init operations on a state machine.
-     * This can include creating the Context for the first time and storing it.
-     * Trigger state machine execution.
+     * Perform init operations on a state machine and starts execution of states which are not dependant on any events.
      * @param stateMachine
      * @return List of states that do not have any event dependencies on them
      */
-    List<State<T>> init(StateMachine<T> stateMachine);
+    public Set<State> initAndStart(StateMachine stateMachine) {
+        Context context = new RAMContext(System.currentTimeMillis(), null); //TODO: set context id, should we need it ?
+        stateMachine.setContext(context);
+        context.buildDependencyMap(stateMachine.getStates());
```

Comment:
done. As context and SM have one-to-one relation, attaching the context to state machine and building dependency graph as part of construction itself.